python/lsst/ts/nightreport/shared_state.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+__all__ = ["create_shared_state", "delete_shared_state", "get_shared_state"]
+
+import logging
+import os
+import urllib
+
+import sqlalchemy as sa
+
+from .create_tables import SITE_ID_LEN, create_nightreport_table
+from .database import NightReportDatabase
+
+_shared_state: None | SharedState = None
+
+
+def get_env(name: str, default: None | str = None) -> str:
+    """Get a value from an environment variable.
+
+    Parameters
+    ----------
+    name
+        The name of the environment variable.
+    default
+        The default value; if None then raise ValueError if absent.
+    """
+    if default is not None and not isinstance(default, str):
+        raise ValueError(f"default={default!r} must be a str or None")
+    value = os.environ.get(name, default)
+    if value is None:
+        raise ValueError(f"You must specify environment variable {name}")
+    return value
+
+
+def create_db_url() -> str:
+    """Create the nightreport database URL from environment variables."""
```

Comment:
Please add an explanation for all parameters and the return type to the doc string.